ta/mk/ta_dev_kit.mk

```diff
@@ -59,6 +59,12 @@ ifeq ($(CFG_TA_MCOUNT),y)
 cppflags$(sm) += -pg
 endif
 
+ifeq ($(CFG_ENCRYPT_TA),y)
+ifneq ($(CFG_REE_FS_TA_BUFFERED),y)
+$(error TA encryption feature is only supported for REE-FS buffered TAs)
```

Comment:
I don't think it is a good thing to test this here. Even if `CFG_REE_FS_TA_BUFFERED` is not set, the "dev kit" can still generate a valid encrypted TA.

Reply:
If you see below, I have enabled `CFG_REE_FS_TA_BUFFERED` to be exported to "TA dev kit" and as I mentioned above `CFG_ENCRYPT_TA` is a TA build time option. So I think this check should work.

Reply:
What makes me uneasy here, is that there is no such thing as a "REE-FS buffered TAs". "Buffered" is how the TEE core decides to load the TA, it is a core configuration, not a TA configuration.

Reply:
I don't see why we need to get far enough to generate a TA dev kit if there's incompatible configuration. We should error out before anything is built.

Reply:
@jforissier 
> "Buffered" is how the TEE core decides to load the TA, it is a core configuration, not a TA configuration.

Agree, but wouldn't this approach be a good idea for user to know that `CFG_REE_FS_TA_BUFFERED` needs to enabled in TEE core during encrypted TA build time? If you have any better ideas then let me know.

@jenswi-linaro 
> I don't see why we need to get far enough to generate a TA dev kit if there's incompatible configuration.

Since we would like to support following configurations:
1. Support to either load an encrypted or a plain TA image from REE-FS. `CFG_REE_FS_TA_BUFFERED` must be enabled.
2. Support only to load plain TA image from REE-FS. `CFG_REE_FS_TA_BUFFERED` is optional which maybe disabled to gain on performance and memory benefit.

So either configuration is valid. And this check is to make user aware about the need to enable `CFG_REE_FS_TA_BUFFERED` if they would like to load an encrypted TA image.

Reply:
@b49020 
I think the  two configurations you're mentioning should not be a compile time choice.  The code can certainly be written so than clear TAs are loaded unbuffered and encrypted TAs are buffered (assuming it's even necessary, which I doubt, since decryption can be done in small steps right?).

Reply:
> The code can certainly be written so than clear TAs are loaded unbuffered and encrypted TAs are buffered

IIUC, you mean to write a wrapper for "REE" ta store that could support loading of encrypted TAs as well. I agree that this could be done. But would you like to still keep "REE [buffered]" ta store?

> assuming it's even necessary, which I doubt, since decryption can be done in small steps right?

Can you please elaborate here? Do you mean TA should be split into parts which are encrypted/decrypted independently? Or do you mean encryption as a whole and decryption in small steps? And what those small steps would be? Lastly how would it be beneficial in terms of performance over a single `memcpy` of TA?

Reply:
Ah, I missed to explore `crypto_authenc_update_payload()` API. Will try to use it instead.